src/plugins/chart_expressions/expression_xy/public/components/reference_lines/reference_line.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC } from 'react';
+import { euiLightVars } from '@kbn/ui-theme';
+import { AnnotationDomainType, LineAnnotation, Position, RectAnnotation } from '@elastic/charts';
+import { FieldFormat } from '@kbn/field-formats-plugin/common';
+import { ReferenceLineConfig, ReferenceLineYConfig } from '../../../common/types';
+import {
+  getBottomRect,
+  getGroupId,
+  getHorizontalRect,
+  getLineAnnotationProps,
+  getSharedStyle,
+} from './utils';
+
+interface ReferenceLineProps {
+  layer: ReferenceLineConfig;
+  paddingMap: Partial<Record<Position, number>>;
+  formatters: Record<'left' | 'right' | 'bottom', FieldFormat | undefined>;
+  axesMap: Record<'left' | 'right', boolean>;
+  isHorizontal: boolean;
+}
+
+const getRectDataValue = (yConfig: ReferenceLineYConfig, formatter: FieldFormat | undefined) => {
+  const { name, value, fill } = yConfig;
+  const isFillAbove = fill === 'above';
+
+  if (yConfig.axisMode === 'bottom') {
+    return getBottomRect(name, isFillAbove, formatter, value);
+  }
+
+  return getHorizontalRect(name, isFillAbove, formatter, value);
+};
+
+export const ReferenceLine: FC<ReferenceLineProps> = ({
+  layer,
+  axesMap,
+  formatters,
+  paddingMap,
+  isHorizontal,
+}) => {
+  const {
+    lineLength,
+    yConfig: [yConfig],
+  } = layer;
+
+  if (!yConfig) {
+    return null;
+  }
+  const { name, value, axisMode } = yConfig;
+
+  // Find the formatter for the given axis
+  const groupId = getGroupId(axisMode);
+
+  const formatter = formatters[groupId || 'bottom'];
+
+  const defaultColor = euiLightVars.euiColorDarkShade;
+
+  const props = getLineAnnotationProps(
+    yConfig,
+    yConfig.textVisibility
+      ? {
+          markerLabel: yConfig.icon ? undefined : name,
+          markerBodyLabel: yConfig.icon ? name : undefined,
+        }
+      : {},
+    axesMap,
+    paddingMap,
+    groupId,
+    isHorizontal
+  );
+
+  const sharedStyle = getSharedStyle(yConfig);
+
+  const dataValuesSample = {
+    dataValue: value,
+    header: name,
+    details: formatter?.convert(value) || value,
+  };
+
+  const dataValues = new Array(lineLength).fill(dataValuesSample);
```

Comment:
Why is it necessary to duplicate the data values in here?

Reply:
https://github.com/elastic/kibana/pull/132192#discussion_r874679791

I don't know the reason for that code to be written in that way. Just found that code could be simplified.

Reply:
this is handling for the case if the table of the reference layer has multiple rows (which can't be configured in Lens today) to render them as multiple lines. It doesn't make sense for the single reference line case.

Reply:
Ok, I see. Let me fix it. But still, before none of the lines from that rows has been taken, except the first one.

Reply:
hah, good point :D I don't think it ever worked correctly.